packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart

```diff
@@ -157,4 +157,39 @@ class InAppPurchaseAndroidPlatformAddition
         .runWithClient((BillingClient client) => client.getBillingConfig());
     return billingConfig.countryCode;
   }
+
+  /// Returns if the caller can use alternative billing only without giving the
+  /// user a choice to use Play billing.
+  ///
+  /// See: https://developer.android.com/reference/com/android/billingclient/api/BillingClient#isAlternativeBillingOnlyAvailableAsync(com.android.billingclient.api.AlternativeBillingOnlyAvailabilityListener)
+  Future<BillingResultWrapper> isAlternativeBillingOnlyAvailable() async {
+    final BillingResultWrapper wrapper =
+        await _billingClientManager.runWithClient((BillingClient client) =>
+            client.isAlternativeBillingOnlyAvailable());
+    return wrapper;
+  }
+
+  /// Shows the alternative billing only information dialog on top of the calling app.
+  ///
+  /// See: https://developer.android.com/reference/com/android/billingclient/api/BillingClient#showAlternativeBillingOnlyInformationDialog(android.app.Activity,%20com.android.billingclient.api.AlternativeBillingOnlyInformationDialogListener)
+  Future<BillingResultWrapper>
+      showAlternativeBillingOnlyInformationDialog() async {
+    final BillingResultWrapper wrapper =
+        await _billingClientManager.runWithClient((BillingClient client) =>
+            client.showAlternativeBillingOnlyInformationDialog());
+    return wrapper;
+  }
+
+  /// Disconnects, Sets AlternativeBillingOnly to true, and recoonects to
+  /// the [BillingClient].
+  ///
+  /// [alternativeBillingOnlyState] true will enable alternative billing only.
+  ///
+  /// Play apis have requirements for when this method can be called.
+  /// See: https://developer.android.com/google/play/billing/alternative/alternative-billing-without-user-choice-in-app
+  Future<void> setAlternativeBillingOnlyState(
```

Comment:
This would also be a strong candidate for a more generic method with a mode enum. (I would say we should either do both this and BillingClientManager, or neither, for consistency.)

Reply:
yep will make an enum and modify it here and and in billing client manager. Then I will probably send down booleans with a documented priority to the java code.